### PR TITLE
release: relnotes: Add Bluetooth release notes for 3.4.0

### PR DESCRIPTION
**Pull request by carlescufi**
As usual, populate the Bluetooth release notes.

### CODE REVIEW

File: doc/releases/release-notes-3.4.rst [hunk 1]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+  * Deprecated the ``BT_DEBUG_LOG`` option. Instead ``BT_LOG`` should be used.
+  * Made the ``BT_LOG`` and ``BT_LOG_LEGACY`` options hidden.
+  * Removed ``BT_DEBUG`` entirely.
+
+
 * Audio
 
+  * Implemented the CAP initiator broadcast audio start, stop and metadata
+    update procedures.
+  * Implemented the CAP unicast audio start, stop and metatda update procedures.
```
Comment (cvinayak) [inline doc/releases/release-notes-3.4.rst:315]:
```suggestion
  * Implemented the CAP unicast audio start, stop and metadata update procedures.
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.4.rst [hunk 2]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+  * Downgraded the severity of select log messages to avoid log flooding.
+  * Separated the handling of LE SC OOB pairing from the legacy OOB logic.
+  * Implemented the Encrypted Advertising Data feature.
+  * Added support for the new Periodic Advertising with Responses (PAwR), both
+    as an advertiser and as a scanner.
+  * Added support for initiating connections from PAwR, as well as receiving
+    connections while synced.
+  * Clarified the behavior that is enabled by the ``BT_PRIVACY`` Kconfig option.
+  * Introduced a new ``seg_recv`` L2CAP API for an applicaiton to receive
+    segmenets directly and manage credits explicitly.
```
Comment (jori-nordic) [inline doc/releases/release-notes-3.4.rst:346]: application, segments

**Review by jori-nordic — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.4.rst [hunk 3]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+      * Opcodes Aggregator Server and Client models.
+    * Proxy Solicitation support including:
+      * Solicitation PDU RPL Configuration Server and Client models.
+      * On-Demand Private Proxy Server and Client models.
+    * Composition Data Page 1 support.
+    * Other Mesh Profile Enhancements.
   * Added experimental support for Mesh Binary Large Object Transfer Model d1.0r04_PRr00 specification.
   * Added experimental support for Mesh Device Firmware Update Model d1.0r04_PRr00 specification.
+  * Fixed multiple profile errata: 11737, 11392, 11627, 11302, 11341
+  * Added async APIs for the Private Beacon Client and Large Comp Data Client.
```
Comment (alxelax) [inline doc/releases/release-notes-3.4.rst:374]:
```
* Added async APIs for the Private Beacon Client and Large Comp Data Client.
* Added support for sending SNB and PRB in parallel.
```
Not a big deal, but there is no Zephyr version with Private Beacon but without parallel SNB and PRB etc.
Is it worth to separate these two lines from their parent functionality above?

File: doc/releases/release-notes-3.4.rst [hunk 4]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+    * Opcodes Aggregator support including:
+      * Opcodes Aggregator Server and Client models.
+    * Proxy Solicitation support including:
+      * Solicitation PDU RPL Configuration Server and Client models.
+      * On-Demand Private Proxy Server and Client models.
+    * Composition Data Page 1 support.
+    * Other Mesh Profile Enhancements.
   * Added experimental support for Mesh Binary Large Object Transfer Model d1.0r04_PRr00 specification.
   * Added experimental support for Mesh Device Firmware Update Model d1.0r04_PRr00 specification.
+  * Fixed multiple profile errata: 11737, 11392, 11627, 11302, 11341
```
Comment (PavelVPV) [inline doc/releases/release-notes-3.4.rst:373]: Could we avoid specifying errata numbers in release notes?

File: doc/releases/release-notes-3.4.rst [hunk 5]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+  * Added async APIs for the Private Beacon Client and Large Comp Data Client.
+  * Added support for sending SNB and PRB in parallel.
```
Comment (PavelVPV) [inline doc/releases/release-notes-3.4.rst:375]: These 2 lines are not needed as this code came in this release with Mesh Protocol d1.1r18 specification support.

File: doc/releases/release-notes-3.4.rst [hunk 6]
```diff
@@ -300,20 +300,103 @@ Architectures
...
+  * Added experimental support for the PSA crypto APIs.
+  * Added a new work queue to store mesh settings, including a new API for
+    storing user data.
```
Comment (PavelVPV) [inline doc/releases/release-notes-3.4.rst:378]: Could you please add:
```
  * Disabled models initialization macros for C++ as they use the Compound Literal feature from C99.
  * Deprecated Health Client and Configuration Client API have been removed.
```

**Review by jhedberg — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Review by alxelax — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**